src/overlays/actors/ovl_En_Pr2/z_en_pr2.c

```diff
@@ -71,48 +92,677 @@ const ActorInit En_Pr2_InitVars = {
+        if (this->unk_1D0 >= this->path->count) {
+            this->unk_1E0 = 2;
+            func_80A74888(this);
+        }
+    }
+
+    this->unk_1EE = Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk_21C);
+    func_80A7436C(this, this->unk_1EE);
+}
+
+void func_80A74888(EnPr2* this) {
+    func_80A74510(this, 0);
+    this->unk_1DA = 2;
+    this->unk_1D8 = 0;
+    Math_Vec3f_Copy(&this->unk_21C, &this->unk_228);
+    this->unk_1D4 = 1;
+    this->actionFunc = func_80A748E8;
+}
+
+void func_80A748E8(EnPr2* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    f32 temp_f12;
+    f32 temp_f2;
+    f32 sqrt;
+    s32 sp4C = 0;
+    s32 sp48 = 0;
+    Vec3f sp3C;
+
+    Math_ApproachF(&this->unk_204, 0.02f, 0.1f, 0.005f);
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_PIRANHA_EXIST - SFX_FLAG);
+
+    if (fabsf(this->actor.world.rot.y - this->unk_1EE) < 200.0f) {
+        sp48 = 1;
+        SkelAnime_Update(&this->skelAnime);
+    }
+
+    if (this->unk_1F4 != 255) {
+        this->actor.speedXZ = 0.0f;
+        Math_SmoothStepToS(&this->unk_1F4, 0, 1, 30, 100);
+        if (this->unk_1F4 < 2) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    } else {
+        switch (this->unk_1E0) {
+            case 1:
+                if (this->unk_1DC == 0) {
+                    sp4C = 1;
+                    func_80A74DEC(this, globalCtx);
+                } else if (!func_80A7429C(this, globalCtx) && (this->unk_1F4 == 255)) {
+                    this->unk_1F4 = 254;
+                }
+                break;
+
+            case 2:
+                if (this->unk_1DE == 0) {
+                    temp_f2 = player->actor.world.pos.x - this->unk_228.x;
+                    temp_f12 = player->actor.world.pos.z - this->unk_228.z;
+                    sqrt = sqrtf(SQ(temp_f2) + SQ(temp_f12));
+
+                    if ((sp48 != 0) && (player->stateFlags1 & 0x8000000) && (sqrt < this->unk_208)) {
+                        sp4C = 1;
+                        func_80A74DEC(this, globalCtx);
+                    }
+                } else {
+                    temp_f2 = this->actor.world.pos.x - this->unk_228.x;
+                    temp_f12 = this->actor.world.pos.z - this->unk_228.z;
+                    sqrt = sqrtf(SQ(temp_f2) + SQ(temp_f12));
+
+                    if (sqrt > 20.0f) {
+                        this->unk_1DE = 5;
+                        sp4C = 1;
+                        this->unk_1DC = 0;
+                        Math_Vec3f_Copy(&this->unk_21C, &this->unk_228);
+                        Math_ApproachF(&this->actor.speedXZ, 3.0f, 0.3f, 0.2f);
+                    }
+                }
+                break;
+        }
+
+        if (sp4C == 0) {
+            this->unk_21C.y = this->actor.world.pos.y;
+            if (this->unk_1DA != 0) {
+                if ((Rand_ZeroOne() < 0.3f) && (this->unk_1D6 == 0)) {
+                    this->unk_1D6 = 1;
```

Comment:
```suggestion
                if ((Rand_ZeroOne() < 0.3f) && !this->unk_1D6) {
                    this->unk_1D6 = true;
```
`unk_1D6` looks boolean, you can change throughout